overlord/snapstate/export_test.go

```diff
@@ -74,3 +74,27 @@ func MockReadInfo(mock func(name string, si *snap.SideInfo) (*snap.Info, error))
 	readInfo = mock
 	return func() { readInfo = snap.ReadInfo }
 }
+
+func RunDoHandler(mgr *SnapManager, kind string, t *state.Task) error {
```

Comment:
I actually find it a bit awkward. It completely ignores all registration of the handler that is actually performed in the real logic and calls a function manually, despite what the real code does. It also fiddles with the task status out of band, which feels slightly curious.

I'd prefer to leverage the real running logic, even if we have to extend the real code a bit. Let's catch up about it live at the sprint.